### DISCUSSION TITLE
segmentation faults with PSATD

### DISCUSSION THREAD

**Discussion by michaelquin** [BLAST-WarpX/warpx#5472] [Q&A]
Hi all,

I’m new to WarpX, but not to PIC in general, so bear with me!

In short, I’m having problems running simulations with the PSATD solver. If I run the minimal working example, i.e. the PICMI script in 
[MWE.zip](https://github.com/user-attachments/files/17817836/MWE.zip), it results in a seg fault before the first step of the simulation.

The backtrace file (also in MWE.zip) seems to confirm that this problem is related to the spectral solver.

However if I use the same script with the Yee solver and Dirichlet boundary conditions, it runs to completion.

Any suggestions would be greatly appreciated. For reference, I'm running WarpX via Anaconda on the HPC Tetralith. If that's sub-optimal, feel free to let me know.

**Comment by oshapoval** [comment]
Hello,

Thank you very much for creating the issue. We do not have the PSATD solver implemented in 1D geometry (but we do have in 2D, 3D and RZ), which is why the segmentation fault occurred. I have created a [PR](https://github.com/ECP-WarpX/WarpX/pull/5500) to add an abort message for cases where 1D PSATD is used, to warn users.
